### DISCUSSION TITLE
umi4 dva持久化怎么实现？

### DISCUSSION THREAD

**Discussion by rpyoyo** [umijs/umi#12197] [Q&A]
umi3+时候使用redux-persist在app.ts里处理持久化，升级到umi4后，好像不行了，是否有相关示例

**Comment by fz6m** [comment]
umi 4 的 dva 、use request 之类的插件都是从 umi 3 直接搬过来的，因为这些旧插件很久都不维护了，所以搬过来只为了兼容以前的项目升级，如果是 umi 3 可以的话，umi 4 也应该可以。

可以自己探究下，如有问题，给一个 umi 3 可以的最小复现看看吧。

注：上文中提到以前的插件都不维护了，因为现代数据流更推荐使用 [react-query](https://umijs.org/docs/max/react-query) / [swr](https://github.com/vercel/swr) 管理请求状态，使用 [jotai](https://github.com/pmndrs/jotai) / [valtio](https://umijs.org/docs/max/valtio) 等管理全局状态。
